cpp/ql/src/semmle/code/cpp/security/BufferWrite.qll

```diff
@@ -78,253 +76,247 @@ abstract class BufferWrite extends Expr
-  }
-  
-  int getParamSize()
-  {
-    if exists(getArgument(2)) then (
-      result = 2
-    ) else (
-      none()
-    )
-  }
-  
-  int getParamSrc()
-  {
-    result = 1
-  }
+class StrCatBW extends BufferWriteCall {
+  StrCatBW() { exists(TopLevelFunction fn | fn = getTarget() and fn instanceof StrcatFunction) }
+
+  int getParamSize() { if exists(getArgument(2)) then result = 2 else none() }
+
+  int getParamSrc() { result = 1 }
 
-  override Type getBufferType()
-  {
+  override Type getBufferType() {
     result = this.getTarget().getParameter(getParamSrc()).getUnspecifiedType()
   }
 
-  override Expr getASource()
-  {
-    result = getArgument(getParamSrc())
-  }
+  override Expr getASource() { result = getArgument(getParamSrc()) }
 
-  override Expr getDest()
-  {
-    result = getArgument(0)
-  }
+  override Expr getDest() { result = getArgument(0) }
 
-  override predicate hasExplicitLimit()
-  {
-    exists(getParamSize())
-  }
+  override predicate hasExplicitLimit() { exists(getParamSize()) }
 
-  override int getExplicitLimit()
-  {
+  override int getExplicitLimit() {
     result = getArgument(getParamSize()).getValue().toInt() * getCharSize()
   }
 
-  override int getMaxData()
-  {
+  override int getMaxData() {
     result = getArgument(getParamSrc()).(AnalysedString).getMaxLength() * getCharSize()
   }
 }
 
 /**
- * A call to a variant of `sprintf`. 
+ * A call to a variant of `sprintf`.
  */
-class SprintfBW extends BufferWriteCall
-{
-  SprintfBW()
-  {
-    exists(TopLevelFunction fn, string name | (fn = getTarget()) and (name = fn.getName()) and (
-      // C sprintf variants
-      (name = "sprintf")                      // sprintf(dst, format, args...)
-      or (name = "vsprintf")                  // vsprintf(dst, format, va_list)
-      or (name = "wsprintf")                  // wsprintf(dst, format, args...)
-      or (name = "vwsprintf")                 // vwsprintf(dst, format, va_list)
-  
-      // Microsoft sprintf variants
-      or (name.regexpMatch("_sprintf_l"))     //  _sprintf_l(dst, format, locale, args...)
-      or (name.regexpMatch("_vsprintf_l"))    //  _vsprintf_l(dst, format, locale, va_list))
-      or (name.regexpMatch("__swprintf_l"))   // __swprintf_l(dst, format, locale, args...)
-      or (name.regexpMatch("__vswprintf_l"))  // __vswprintf_l(dst, format, locale, va_list)
-    ))
-  }
-
-  override Type getBufferType()
-  {
+class SprintfBW extends BufferWriteCall {
+  SprintfBW() {
+    exists(TopLevelFunction fn, string name | fn = getTarget() and name = fn.getName() |
+      /*
+       * C sprintf variants
+       */
+
+      // sprintf(dst, format, args...)
+      name = "sprintf"
+      or
+      // vsprintf(dst, format, va_list)
+      name = "vsprintf"
+      or
+      // wsprintf(dst, format, args...)
+      name = "wsprintf"
+      or
+      // vwsprintf(dst, format, va_list)
+      name = "vwsprintf"
+      or
+      /*
+       * Microsoft sprintf variants
+       */
+
```

Comment:
I think we want the space _after_ the comment because it suggests that the comment isn't only about the next clause after it. I agree it looks funny that there's no space _before_ it, but at least it's consistent with the "C snprintf variants" comment above.

I'll add colons after the four comments that are like this. Does that solve the problem well enough?

Reply:
Yeah, it's not perfect but I think it will do.